source/Fleet.cpp

```diff
@@ -88,7 +88,24 @@ void Fleet::Load(const DataNode &node)
 		else if(key == "names" && hasValue)
 			names = GameData::Phrases().Get(child.Token(1));
 		else if(key == "fighters" && hasValue)
+		{
 			fighterNames = GameData::Phrases().Get(child.Token(1));
+			child.PrintTrace("Warning: Deprecated use of \"fighters\" <names>."
+				" Use \"names\" <names> node under \"fighters\" instead.");
+		}
+		else if(key == "fighters" && child.HasChildren())
+		{
+			for(const DataNode &grand : child)
+			{
+				const string &fighterKey = grand.Token(0);
+				if(fighterKey == "names" && grand.Size() >= 2)
+					fighterNames = GameData::Phrases().Get(grand.Token(1));
+				else if(fighterKey == "personality")
+					fighterPersonality.Load(grand);
+				else
+					grand.PrintTrace("Skipping unrecognized attribute under fighters:");
```

Comment:
do we normally add an "under [parent node]"? I thought this automatically gets added by the print trace

Reply:
You are correct, the print trace makes it clear what the parent node is without needing to specify here.